src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/service/cessations/BeneficialOwnerCessationService.java

```diff
@@ -0,0 +1,111 @@
+    cessations.addAll(returnCorporateEntityBeneficialOwnerCessations(overseasEntitySubmissionDto, combinedBoData));
+    cessations.addAll(returnLegalPersonBeneficialOwners(overseasEntitySubmissionDto, combinedBoData));
+    return cessations;
+  }
+
+  private List<IndividualBeneficialOwnerCessation> returnIndividualBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var beneficialOwnersIndividual = overseasEntitySubmissionDto.getBeneficialOwnersIndividual();
+    return beneficialOwnersIndividual.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnIndividualBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<CorporateEntityBeneficialOwnerCessation> returnCorporateEntityBeneficialOwnerCessations(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersCorporate();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnCorporateEntityBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private List<LegalPersonBeneficialOwnerCessation> returnLegalPersonBeneficialOwners(
+          OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+          Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+    var returned = overseasEntitySubmissionDto.getBeneficialOwnersGovernmentOrPublicAuthority();
+    return returned.stream()
+            .filter(beneficialOwner -> (beneficialOwner.getCeasedDate() != null))
+            .filter(beneficialOwner -> (beneficialOwner.getChipsReference() != null))
+            .map(beneficialOwner -> returnLegalPersonBeneficialOwnerCessation(beneficialOwner, combinedBoData))
+            .collect(Collectors.toList());
+  }
+
+  private IndividualBeneficialOwnerCessation returnIndividualBeneficialOwnerCessation(
+      BeneficialOwnerIndividualDto bo, Map<String, Pair<PscApi, PrivateBoDataApi>> combinedBoData) {
+
+    var hashedId = bo.getChipsReference();
+    var publicPrivateBoPair = Optional.ofNullable(combinedBoData.get(hashedId))
+            .orElseThrow(() -> new IllegalArgumentException(NO_PAIR_FOUND));
+
+    var privateBoData = publicPrivateBoPair.getRight();
+    var appointmentId = Optional.ofNullable(privateBoData).map(PrivateBoDataApi::getPscId).orElse(null);
```

Comment:
We can't just silently pass on a null. Skip the cessation as above after logging an error.

Reply:
Addressed